components/script/script_thread.rs

```diff
@@ -881,6 +900,29 @@ impl ScriptThread {
         })
     }
 
+    pub fn get_currently_not_fully_active_document_ids() -> HashSet<PipelineId> {
```

Comment:
Could you explain why a `HashSet` of non-fully active documents is needed here? Aren't there less full active documents than non-fully active ones at any given time?